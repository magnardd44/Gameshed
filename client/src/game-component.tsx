import * as React from 'react';
import { Component } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, reviewService, Game, Review } from './services';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

export class GameCard extends Component {
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: [],
    genre_id: 0,
    platform: [],
    game_description: '',
  };
  description =
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco...';
  render() {
    return (
      <Card title={this.game.game_title}>
        <h6 className="card-subtitle mb-2 text-muted">
          Terningkast:
          <ThumbNail small img="https://cdn-icons-png.flaticon.com/512/220/220725.png"></ThumbNail>
        </h6>
        <Row>
          <Column width={2}>
            <ThumbNail img="https://cdn-icons-png.flaticon.com/512/686/686589.png"></ThumbNail>
          </Column>
          <Column width={6}>
            {this.game.game_description}
            <Linebreak></Linebreak>
          </Column>
          <Column width={2}></Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Sjanger: {this.game.genre}</Column>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <Column>Platform: {this.game.platform}</Column>
        </Row>
      </Card>
    );
  }
  /** return axios
   */
  mounted() {
    gameService.getAll().then((result) => {
      this.game = result[0];
      console.log(this.game);
    });
  }
}
