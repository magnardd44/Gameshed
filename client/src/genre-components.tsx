import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, Game } from './services/game-services';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import { Review, reviewService } from './services/review-service';

const history = createHashHistory();

export class Category extends Component {
  genres: Genre[] = [];
  reviews: Review[] = [];
  games: Game[] = [];
  game: Game = {
    game_id: 0,
    game_title: '',
    genre: 0,
    genre_id: 0,
    platform: 0,
    game_description: '',
    igdb_id: 0,
    genres: [],
    platforms: [],
  };
  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
  };
  state = { isHidden: false };
  genreCall(id: number) {
    reviewService
      .getGenre(id)
      .then((data) => {
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
    this.setState({ isHidden: false });
  }
  render() {
    return (
      <Container>
        Velg kategori/sjanger:
        <Row>
          <ColumnCentre width={12} smwidth={4} mdwidth={2}>
            {this.genres.map((genre) => (
              <CategoryCard
                key={genre.genre_id}
                img={
                  genre.genre_img
                    ? 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                      genre.genre_img +
                      '.png'
                    : 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                      genre.genre_name +
                      '.png'
                }
              >
                <Button.Success onClick={() => this.genreCall(genre.genre_id)}>
                  {genre.genre_name}
                </Button.Success>
              </CategoryCard>
            ))}
          </ColumnCentre>
          <ColumnCentre>
            {this.state.isHidden ? null : (
              <div className="sticky-top">
                {this.reviews.map((review, index) => (
                  <Row key={index}>
                    <Column>{review.game_title}</Column>
                    <Column>
                      <NavLink to={'/publishedReviews/' + review.review_id}>
                        {review.review_title}
                      </NavLink>
                    </Column>
                    <Column>{review.rating}</Column>
                  </Row>
                ))}
              </div>
            )}
          </ColumnCentre>
        </Row>
      </Container>
    );
  }
  mounted() {
    genreService
      .getAll()
      .then((genres) => {
        this.genres = genres;
      })
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
  }
}
