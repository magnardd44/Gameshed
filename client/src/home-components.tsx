import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import userService from './services/user-service';

export const history = createHashHistory();

export class Home extends Component {
  render() {
    return (
      <Container textalign={'center'}>
        <Row>
          <ColumnCentre width={6}>
            <Card title={'Søk etter spill'}>
              <Button.Success
                onClick={() => {
                  history.push('/search');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
          <ColumnCentre width={6}>
            <Card title={'Søk etter anmeldelser'}>
              <Button.Success
                onClick={() => {
                  history.push('/reviews');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <ColumnCentre width={userService.token ? 12 : 6}>
            <Card title={'Legg til nytt spill'}>
              <Button.Success
                onClick={() => {
                  history.push('/addGame');
                }}
              >
                Legg til spill
              </Button.Success>
            </Card>
          </ColumnCentre>
          {userService.token ? (
            ''
          ) : (
            <ColumnCentre width={6}>
              <Card title="Registrer ny konto">
                <Button.Success
                  onClick={() => {
                    history.push('/user');
                  }}
                >
                  Registrer deg
                </Button.Success>
              </Card>
            </ColumnCentre>
          )}
        </Row>
      </Container>
    );
  }
}

export class ReviewHome extends Component {
  render() {
    return (
      <Container textalign={'center'}>
        <Row>
          <ColumnCentre width={6}>
            <Card title={'Sorter etter sjanger'}>
              <Button.Success
                onClick={() => {
                  history.push('/reviews-by-genre');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
          <ColumnCentre width={6}>
            <Card title={'Sorter etter plattform'}>
              <Button.Success
                onClick={() => {
                  history.push('/reviews-by-platform');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <ColumnCentre width={6}>
            <Card title={'Topp ti anmeldelser'}>
              <Button.Success
                onClick={() => {
                  history.push('/reviews-by-stars');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
          <ColumnCentre width={6}>
            <Card title={'Siste ti anmeldelser?'}>
              <Button.Success
                onClick={() => {
                  history.push('/reviews-by-date');
                }}
              >
                Søk
              </Button.Success>
            </Card>
          </ColumnCentre>
        </Row>
        <Linebreak></Linebreak>
        <Row>
          <ColumnCentre width={12}>
            <Card title={'Alle anmeldelser'}>
              <Button.Success
                onClick={() => {
                  history.push('/publishedReviews');
                }}
              >
                Alle anmeldelser
              </Button.Success>
            </Card>
          </ColumnCentre>
        </Row>
      </Container>
    );
  }
}
