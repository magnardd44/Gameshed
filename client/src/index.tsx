import * as React from 'react';
import { Component } from 'react-simplified';
import ReactDOM from 'react-dom';
import {
  Card,
  Alert,
  Row,
  Form,
  Column,
  Button,
  NavBar,
  Header,
  Container,
  ColumnCentre,
  SearchResult,
} from './widgets';
import axios from 'axios';
import { HashRouter, Route } from 'react-router-dom';
import { AddReview, PublishReview } from './review-components';
import { Hash } from 'crypto';
import { createHashHistory } from 'history';

axios.defaults.baseURL = 'http://localhost:3000/api/v2';

const history = createHashHistory();

class NavHeader extends Component {
  render() {
    return (
      <div>
        <NavBar brand="GameShed">
          <NavBar.Link to="/games">Games</NavBar.Link>
        </NavBar>
        <Header></Header>
      </div>
    );
  }
}

class Search extends Component {
  input = '';
  render() {
    return (
      <>
        <Container textalign={'centre'}>
          <Row>
            <ColumnCentre width={6} offset={2}>
              <Form.Input
                type={this.input}
                value={this.input}
                placeholder="Søk etter et spill"
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </ColumnCentre>
            <ColumnCentre width={2}>
              <Button.Success
                onClick={() => {
                  this.search();
                }}
              >
                Søk
              </Button.Success>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  search() {
    console.log('search');
    history.push('/results');
  }
}

class SearchListings extends Component {
  render() {
    return (
      <Container>
        Søkeresultater:
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
        <SearchResult></SearchResult>
      </Container>
    );
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <NavHeader />
      <Route exact path="/" component={Search}></Route>
      <Route exact path="/results" component={SearchListings}></Route>
      <Route exact path="/addReview" component={AddReview} />
      <Route exact path="/publishReview/:id" component={PublishReview} />
    </div>
  </HashRouter>,
  document.getElementById('root')
);

/*
class Start extends Component {
  input = '';
  stdout = '';
  stderr = '';
  errCode: number | null = null;
}



  render() {
    return (
      <>
        <Card title="GameShed">
          <Row>
            <Column>
              <Form.Label>Søk:</Form.Label>
              <Form.Textarea
                value={this.input}
                onChange={(event) => (this.input = event.currentTarget.value)}
              />
            </Column>
          </Row>
          <Button.Success
            onClick={() => {
              axios
                .post<{ exitStatus: number; stdout: string; stderr: string }>('/run', {
                  language: 'js',
                  source: this.input,
                })
                .then((response) => {
                  this.errCode = response.data.exitStatus;
                  this.stdout = response.data.stdout;
                  this.stderr = response.data.stderr;
                })
                .catch((error: Error) => Alert.danger('Could not run app.js: ' + error.message));
            }}
          >
            Søk etter spill
          </Button.Success>
        </Card>
        <Card title="Standard output">{this.stdout}</Card>
        <Card title="Standard error">{this.stderr}</Card>
        <Card title="Errorcode">{this.errCode}</Card>
      </>
    );
  }
}


const root = document.getElementById('root');
if (root)
  ReactDOM.render(
    <HashRouter>
      <Alert />

      <Route exact path="/" component={Start} />
      
    </HashRouter>,
    root
  );
 
 */
