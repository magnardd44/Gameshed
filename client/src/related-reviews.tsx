import * as React from 'react';
import { Component } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  ReviewCard,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, Game } from './services/game-services';
import { gameService2, Game2 } from './services/game-services';
import { Genre, genreService } from './services/genre-service';
import { Review, reviewService } from './services/review-service';
import { createHashHistory } from 'history';
import { platform } from 'os';
import { Platform, platformService } from './services/platform-service';
import axios from 'axios';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student

export class RelatedReviews extends Component<{ game_id?: number; genre_id?: number }> {
  render() {
    return (
      <Container>
        <Row>
          <h5>Relaterte anmeldelser</h5>
          <ColumnCentre width={4}>
            <ReviewCard
              title={reviewService.relatedReview.review_title}
              subtitle={reviewService.relatedReview.game_title}
              terningkast={reviewService.relatedReview.rating}
              relevanse={reviewService.relatedReview.likes}
              text={reviewService.relatedReview.text}
            ></ReviewCard>
          </ColumnCentre>
        </Row>
        <Row>
          {reviewService.relatedReviews.map((review) => {
            <ColumnCentre width={4}>
              <ReviewCard
                title={review.review_title}
                subtitle={review.game_title}
                terningkast={review.rating}
                relevanse={review.likes}
                text={review.text}
              ></ReviewCard>
            </ColumnCentre>;
          })}
        </Row>
      </Container>
    );
  }

  mounted() {
    reviewService.getGenre(1).then((reviews) => {
      reviewService.relatedReviews = reviews;
      console.log('reviews' + reviewService.relatedReviews[0].review_title);
    });

    reviewService
      .get(1)
      .then((review) => {
        reviewService.relatedReview = review;
        console.log('reviews' + reviewService.relatedReviews[0].review_title);
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
  }
}
