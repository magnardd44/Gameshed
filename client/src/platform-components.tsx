import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
  ReviewCard,
  DropDown,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { gameService, Game } from './services/game-services';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import { platformService } from './services/platform-service';
import { Review, reviewService } from './services/review-service';
import { RelatedReviews } from './related-reviews';

const history = createHashHistory();

//Renders overview of published reviews based on genre
export class Platform extends Component {
  reviews: Review[] = [];
  games: Game[] = [];
  state = { isHidden: true };

  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
    likes: 0,
  };

  platformCall(id: number) {
    reviewService
      .getPlatform(id)
      .then((data) => {
        this.setState({ isHidden: false });
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
  }

  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du søke anmeldelser etter platform</h3>
          <div className="text-center">
            <Button.Success
              small
              onClick={() => {
                history.push('/reviews-by-genre');
              }}
            >
              Klikk her
            </Button.Success>{' '}
            for å søke etter sjanger
          </div>
          <Linebreak></Linebreak>
          <h5>Velg platform:</h5>
          <Row>
            <ColumnCentre width={6} smwidth={4} mdwidth={2}>
              <Card title="Play Station 4">
                <Button.Success onClick={() => this.platformCall(145)}>Vis</Button.Success>
              </Card>
              <Card title="Play Station 5">
                <Button.Success onClick={() => this.platformCall(146)}>Vis</Button.Success>
              </Card>
              <Card title="Nintendo Switch">
                <Button.Success onClick={() => this.platformCall(119)}>Vis</Button.Success>
              </Card>
              <Card title="XBox One">
                <Button.Success onClick={() => this.platformCall(186)}>Vis</Button.Success>
              </Card>
              <Card title="XBox 360">
                <Button.Success onClick={() => this.platformCall(185)}>Vis</Button.Success>
              </Card>
              <Card title="PC - Microsoft">
                <Button.Success onClick={() => this.platformCall(126)}>Vis</Button.Success>
              </Card>
              <Card title="Gameboy">
                <Button.Success onClick={() => this.platformCall(89)}>Vis</Button.Success>
              </Card>
              <Card title="Mac">
                <Button.Success onClick={() => this.platformCall(100)}>Vis</Button.Success>
              </Card>
              <Card title="IOS">
                <Button.Success onClick={() => this.platformCall(98)}>Vis</Button.Success>
              </Card>
              <Card title="Nintendo DS">
                <Button.Success onClick={() => this.platformCall(114)}>Vis</Button.Success>
              </Card>
              <Card title="Andre">
                <Button.Success onClick={() => this.platformCall(101 + 102)}>Vis</Button.Success>
              </Card>
            </ColumnCentre>
            <ColumnCentre>
              {' '}
              {this.state.isHidden ? null : (
                <>
                  <Row>
                    <Column>Resultater</Column>
                  </Row>
                  {this.reviews.map((review, index) => (
                    <Row key={index}>
                      <ReviewCard
                        title={review.review_title}
                        subtitle={review.game_title}
                        terningkast={review.rating}
                        relevanse={review.likes}
                        text={review.text}
                      >
                        <Button.Success
                          small
                          onClick={() => {
                            history.push('/publishedReviews/' + review.review_id);
                          }}
                        >
                          Les mer
                        </Button.Success>
                      </ReviewCard>
                    </Row>
                  ))}
                </>
              )}
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }

  mounted() {
    reviewService
      .getPublishedReviews()
      .then((reviews) => (this.reviews = reviews))
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
  }
}

/**
 * 
export class Platform extends Component {
  platforms: Platform[] = [];
  genres: Genre[] = [];
  reviews: Review[] = [];

  review: Review = {
    review_id: 0,
    game_id: 0,
    game_title: '',
    review_title: '',
    text: '',
    user_id: 0,
    rating: 0,
    published: false,
    genre_id: 0,
    relevant: 0,
    platform_id: 0,
    likes: 0,
  };
  state = { isHidden: false };
  platformCall(id: number) {
    reviewService
      .getPlatform(id)
      .then((data) => {
        //this.setState({ isHidden: false });
        console.log(data);
        this.reviews = data;
      })
      .catch((error) => Alert.danger('Error retrieving reviews: ' + error.message));
  }
  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du søke anmeldelser etter platform</h3>
          <div className="text-center">
            <Button.Success
              small
              onClick={() => {
                history.push('/reviews-by-genre');
              }}
            >
              Klikk her
            </Button.Success>{' '}
            for å søke etter sjanger
          </div>
          <Linebreak></Linebreak>
          <h5>Velg platform:</h5>

          <Row>
            <Column width={1}>
              <Card title="Play Station 4">
                <Button.Success onClick={() => this.platformCall(145)}>Open</Button.Success>
              </Card>
            </Column>
            <ColumnCentre width={6} smwidth={4} mdwidth={2}>
              {this.genres.map((genre) => (
                <CategoryCard
                  key={genre.genre_id}
                  img={
                    genre.genre_img
                      ? 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                        genre.genre_img +
                        '.png'
                      : 'https://helenaagustsson.github.io/INFT2002-images/images/' +
                        genre.genre_name +
                        '.png'
                  }
                >
                  <Button.Success
                    onClick={() => {
                      this.platformCall(genre.genre_id);
                    }}
                  >
                    {genre.genre_name}
                  </Button.Success>
                </CategoryCard>
              ))}
            </ColumnCentre>

            <ColumnCentre>
              {this.state.isHidden ? null : (
                <div className="sticky-top">
                  <div>Resultater:</div>
                  {this.reviews.map((review, index) => (
                    <Row key={index}>
                      <ReviewCard
                        title={review.review_title}
                        subtitle={review.game_title}
                        terningkast={review.rating}
                        relevanse={review.likes}
                        text={review.text}
                      >
                        <Button.Success
                          small
                          onClick={() => {
                            history.push('/publishedReviews/' + review.review_id);
                          }}
                        >
                          Les mer
                        </Button.Success>
                      </ReviewCard>
                    </Row>
                  ))}
                </div>
              )}
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  mounted() {
    reviewService
      .getPublishedReviews()
      .then((reviews) => (this.reviews = reviews))
      .catch((error) => Alert.danger('Error getting reviews: ' + error.message));
    //platformService.getAll().then((platforms)=>{this.platforms = platforms})
  }
}

 * 
 */
