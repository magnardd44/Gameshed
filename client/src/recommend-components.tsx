import * as React from 'react';
import { Component, sharedComponentData } from 'react-simplified';
import {
  Alert,
  Card,
  Row,
  Column,
  Form,
  Button,
  ThumbNail,
  Container,
  ColumnCentre,
  Linebreak,
  FormContainer,
  FormGroup,
  CategoryCard,
  ReviewCard,
} from './widgets';
import { NavLink } from 'react-router-dom';
import { Genre } from './services/genre-service';
import { createHashHistory } from 'history';
import axios from 'axios';
import { genreService } from './services/genre-service';
import { Review, reviewService } from './services/review-service';
import { RelatedReviews } from './related-reviews';

export const history = createHashHistory();

export class TopTenStars extends Component {
  reviews: Review[] = [];
  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du se de best likte anmeldelsene</h3>
          <Linebreak></Linebreak>
          <Row>
            <ColumnCentre>
              <ReviewList reviews={this.reviews}></ReviewList>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  mounted() {
    reviewService
      .getTopTen()
      .then((reviews) => {
        this.reviews = reviews;
      })
      .catch((error) => Alert.danger('Error getting top ten reviews: ' + error.message));
  }
}

export class LastTen extends Component {
  reviews: Review[] = [];
  render() {
    return (
      <>
        <Container>
          <h3 className="text-center">Her kan du se de siste publiserte anmeldelsene</h3>
          <Linebreak></Linebreak>
          <Row>
            <ColumnCentre>
              <ReviewList reviews={this.reviews}></ReviewList>
            </ColumnCentre>
          </Row>
        </Container>
      </>
    );
  }
  mounted() {
    reviewService
      .getLastTen()
      .then((reviews) => {
        this.reviews = reviews;
      })
      .catch((error) => Alert.danger('Error getting top ten reviews: ' + error.message));
  }
}

export class ReviewList extends Component<{ reviews: Review[] }> {
  render() {
    return (
      <div className="sticky-top">
        {this.props.reviews.length == 0 ? (
          <Row>
            <Column>Ingen resultater</Column>
          </Row>
        ) : (
          this.props.reviews.map((review, index) => (
            <Row key={index}>
              <ReviewCard
                title={review.review_title}
                subtitle={review.game_title}
                terningkast={review.rating}
                relevanse={review.likes}
                text={review.text}
              >
                <Button.Success
                  small
                  onClick={() => {
                    history.push('/publishedReviews/' + review.review_id);
                  }}
                >
                  Les mer
                </Button.Success>
              </ReviewCard>
            </Row>
          ))
        )}
      </div>
    );
  }
}
