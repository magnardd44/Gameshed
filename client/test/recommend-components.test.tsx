import * as React from 'react';
//import ReactDOM from 'react-dom';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import { shallow } from 'enzyme';
import { history, TopTenStars, LastTen, ReviewList } from '../src/recommend-components';
import {
  ReviewCard,
  ColumnCentre,
  Form,
  Card,
  Alert,
  Button,
  Container,
  Column,
} from '../src/widgets';

jest.setTimeout(1000);

const mockAdapter = new MockAdapter(axios);

const completeReviews = [
  {
    review_id: 1,
    game_id: 1,
    game_title: 'abc',
    review_title: 'def',
    text: 'ghi',
    user_id: 1,
    rating: 4,
    published: true,
    genre_id: 1,
    platform_id: 1,
    relevant: 1,
    likes: 1,
    user_nickname: 'jkl',
  },
  {
    review_id: 2,
    game_id: 2,
    game_title: 'abc',
    review_title: 'def',
    text: 'ghi',
    user_id: 2,
    rating: 2,
    published: true,
    genre_id: 2,
    platform_id: 2,
    relevant: 1,
    likes: 2,
    user_nickname: 'jkl',
  },
];

describe('TopTenStars component', () => {
  test('TopTenStars default', () => {
    mockAdapter.onGet('/reviews/topTen/').reply(200, []);

    //@ts-ignore
    const wrapper = shallow(<TopTenStars />);
    expect(wrapper).toMatchSnapshot();
  });

  test('TopTenStars with reviews', (done) => {
    mockAdapter.onGet('/reviews/topTen/').reply(200, completeReviews);

    //@ts-ignore
    const wrapper = shallow(<TopTenStars />);

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      //@ts-ignore
      expect(wrapper.instance().reviews).toHaveLength(2);
      done();
    });
  });

  test('TopTenStars without reviews', (done) => {
    const spy = jest.spyOn(Alert, 'danger');
    mockAdapter.onGet('/reviews/topTen/').reply(500);

    //@ts-ignore
    const wrapper = shallow(<TopTenStars />);

    setTimeout(() => {
      expect(spy).toBeCalled();
      done();
    });
  });
});

describe('LastTen component', () => {
  test('LastTen default', () => {
    mockAdapter.onGet('/reviews/topTen/').reply(200, []);

    //@ts-ignore
    const wrapper = shallow(<LastTen />);
    expect(wrapper).toMatchSnapshot();
  });

  test('LastTen with reviews', (done) => {
    mockAdapter.onGet('/reviews/lastTen/').reply(200, completeReviews);

    //@ts-ignore
    const wrapper = shallow(<LastTen />);

    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      //@ts-ignore
      expect(wrapper.instance().reviews).toHaveLength(2);
      done();
    });
  });

  test('LastTen without reviews', (done) => {
    const spy = jest.spyOn(Alert, 'danger');
    mockAdapter.onGet('/reviews/lastTen/').reply(500);

    //@ts-ignore
    const wrapper = shallow(<LastTen />);

    setTimeout(() => {
      expect(spy).toBeCalled();
      done();
    });
  });
});

describe('ReviewList component', () => {
  test('ReviewList default', () => {
    //@ts-ignore
    const wrapper = shallow(<ReviewList reviews={[]} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(ReviewCard)).toHaveLength(0);
  });

  test('ReviewList with reviews', () => {
    //@ts-ignore
    const wrapper = shallow(<ReviewList reviews={completeReviews} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(ReviewCard)).toHaveLength(2);
  });

  test('Click on review', () => {
    //@ts-ignore
    const wrapper = shallow(<ReviewList reviews={completeReviews} />);

    const spy = jest.spyOn(history, 'push');

    wrapper.find(Button.Success).at(0).simulate('click');

    expect(spy).toBeCalled();
  });
});
