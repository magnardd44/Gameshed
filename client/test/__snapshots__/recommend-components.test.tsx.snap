// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastTen component LastTen default 1`] = `
<Fragment>
  <Container>
    <h3
      className="text-center"
    >
      Her kan du se de siste publiserte anmeldelsene
    </h3>
    <Linebreak />
    <Row>
      <ColumnCentre>
        <ReviewList
          reviews={Array []}
        />
      </ColumnCentre>
    </Row>
  </Container>
</Fragment>
`;

exports[`LastTen component LastTen with reviews 1`] = `
<Fragment>
  <Container>
    <h3
      className="text-center"
    >
      Her kan du se de siste publiserte anmeldelsene
    </h3>
    <Linebreak />
    <Row>
      <ColumnCentre>
        <ReviewList
          reviews={
            Array [
              Object {
                "game_id": 1,
                "game_title": "abc",
                "genre_id": 1,
                "likes": 1,
                "platform_id": 1,
                "published": 1,
                "rating": 4,
                "relevant": 1,
                "review_id": 1,
                "review_title": "def",
                "text": "ghi",
                "user_id": 1,
                "user_nickname": "jkl",
              },
              Object {
                "game_id": 2,
                "game_title": "abc",
                "genre_id": 2,
                "likes": 2,
                "platform_id": 2,
                "published": 1,
                "rating": 2,
                "relevant": 1,
                "review_id": 2,
                "review_title": "def",
                "text": "ghi",
                "user_id": 2,
                "user_nickname": "jkl",
              },
            ]
          }
        />
      </ColumnCentre>
    </Row>
  </Container>
</Fragment>
`;

exports[`ReviewList component ReviewList default 1`] = `
<div
  className="sticky-top"
>
  <Row>
    <Column>
      Ingen resultater
    </Column>
  </Row>
</div>
`;

exports[`ReviewList component ReviewList with reviews 1`] = `
<div
  className="sticky-top"
>
  <Row
    key="0"
  >
    <ReviewCard
      relevanse={1}
      subtitle="abc"
      terningkast={4}
      text="ghi"
      title="def"
    >
      <ButtonSuccess
        onClick={[Function]}
        small={true}
      >
        Les mer
      </ButtonSuccess>
    </ReviewCard>
  </Row>
  <Row
    key="1"
  >
    <ReviewCard
      relevanse={2}
      subtitle="abc"
      terningkast={2}
      text="ghi"
      title="def"
    >
      <ButtonSuccess
        onClick={[Function]}
        small={true}
      >
        Les mer
      </ButtonSuccess>
    </ReviewCard>
  </Row>
</div>
`;

exports[`TopTenStars component TopTenStars default 1`] = `
<Fragment>
  <Container>
    <h3
      className="text-center"
    >
      Her kan du se de best likte anmeldelsene
    </h3>
    <Linebreak />
    <Row>
      <ColumnCentre>
        <ReviewList
          reviews={Array []}
        />
      </ColumnCentre>
    </Row>
  </Container>
</Fragment>
`;

exports[`TopTenStars component TopTenStars with reviews 1`] = `
<Fragment>
  <Container>
    <h3
      className="text-center"
    >
      Her kan du se de best likte anmeldelsene
    </h3>
    <Linebreak />
    <Row>
      <ColumnCentre>
        <ReviewList
          reviews={
            Array [
              Object {
                "game_id": 1,
                "game_title": "abc",
                "genre_id": 1,
                "likes": 1,
                "platform_id": 1,
                "published": 1,
                "rating": 4,
                "relevant": 1,
                "review_id": 1,
                "review_title": "def",
                "text": "ghi",
                "user_id": 1,
                "user_nickname": "jkl",
              },
              Object {
                "game_id": 2,
                "game_title": "abc",
                "genre_id": 2,
                "likes": 2,
                "platform_id": 2,
                "published": 1,
                "rating": 2,
                "relevant": 1,
                "review_id": 2,
                "review_title": "def",
                "text": "ghi",
                "user_id": 2,
                "user_nickname": "jkl",
              },
            ]
          }
        />
      </ColumnCentre>
    </Row>
  </Container>
</Fragment>
`;
